src/lib/autoMode.js

```diff
@@ -1,29 +1,57 @@
 import appState from './appState';
 import presets from './autoPresets';
+import generateFunction from './generate-equation';
+import wrapVectorField from './wrapVectorField';
 
-let delayTime, incomingPresetsQueue, scene, scheduledUpdate;
+let delayTime, incomingPresetsQueue, scene, scheduledUpdate, autoSource;
 
 export function initAutoMode(_scene) {
   scene = _scene;
 
-  const auto = appState.getQS().get('auto');
-  if (!auto) {
-    return;
+  const qs = appState.getQS();
+
+  autoSource = qs.get('autosource');
+  if (!['presets', 'generator', 'both'].includes(autoSource)) {
```

Comment:
Just a note: this would probably break IE 11, but I think I already broke it with Promises. Might be simpler to just `!autosorce || autosource.match(/(presets|generator|both)/))`

Reply:
Good point... Or even `['presets', 'generator', 'both'].indexOf(autoSource) !== -1`.

I sometimes lose track of what magic Babel does and what it doesn't.

Thank you for working with me on these patches!